### DISCUSSION TITLE
[Feature Request] Back/Forward button navigation for carousel

### DISCUSSION THREAD

**Discussion by thdoan** [saadeghi/daisyui#1682] [Ideas / feature requests]
It's cool that the carousel component lets you navigate to previous/next slides using #anchors. However, the UX can be improved if the user is able to navigate to the previous slide using the Back button. Currently the experience is such, for example, if you're on #slide2 and you hit the Back button, the anchor in the URL changes to #slide1 but you're still on slide 2.

**Comment by saadeghi** [comment]
You need to do that using JS because daisyUI is just a CSS file and keyboard navigation is not possible without JS.

**Reply by thdoan** [reply]
Thanks, I managed to resolve this with JS. For those who find themselves in the same boat, you can do something like this:

```
function onHashChange() {
  const hash = location.hash;

  if (hash) {
    document.querySelector(hash)?.scrollIntoView();
  }
}

addEventListener('hashchange', onHashChange);
```
